sdk/trace/batch_span_processor_test.go

```diff
@@ -633,3 +644,235 @@ func TestBatchSpanProcessorConcurrentSafe(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestBatchSpanProcessorMetricsDisabled(t *testing.T) {
+	t.Setenv("OTEL_GO_X_SELF_OBSERVABILITY", "false")
+	tp := basicTracerProvider(t)
+	reader := sdkmetric.NewManualReader()
+	meterProvider := sdkmetric.NewMeterProvider(sdkmetric.WithReader(reader))
+	otel.SetMeterProvider(meterProvider)
+	me := newBlockingExporter()
+	t.Cleanup(func() { assert.NoError(t, me.Shutdown(context.Background())) })
+	bsp := NewBatchSpanProcessor(
+		me,
+		// Make sure timeout doesn't trigger during the test.
+		WithBatchTimeout(time.Hour),
+		WithMaxQueueSize(2),
+		WithMaxExportBatchSize(2),
+	)
+	tp.RegisterSpanProcessor(bsp)
+
+	tr := tp.Tracer("TestBatchSpanProcessorMetricsDisabled")
+	// Generate 2 spans, which export and block during the export call.
+	generateSpan(t, tr, testOption{genNumSpans: 2})
+	me.waitForSpans(2)
```

Comment:
(maybe nit) This can wait forever if the condition is not met, shouldn't we give some timeout/deadline?

Reply:
done in https://github.com/open-telemetry/opentelemetry-go/pull/6393/commits/bd6e7b2243257e16595391c86885feae74dd6831